gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java

```diff
@@ -0,0 +1,389 @@
+    DOMAIN("domain:");
+
+    private final String prefix;
+
+    MemberType(String prefix) {
+      this.prefix = prefix;
+    }
+
+    String prefix() {
+      return prefix;
+    }
+  }
+
+  public enum RoleType {
+    OWNER,
+    EDITOR,
+    VIEWER;
+  }
+
+  /**
+   * Represents a member belonging to an IAM policy binding
+   */
+  public static final class Member implements Serializable {
+
+    private static final long serialVersionUID = 6496912037577986137L;
+    private final MemberType memberType;
+    private final String emailOrDomain;
+
+    Member(MemberType memberType, String emailOrDomain) {
+      this.memberType = memberType;
+      this.emailOrDomain = emailOrDomain;
+    }
+
+    public static Member allUsers() {
+      throw new UnsupportedOperationException(
+          "Google Cloud Resource Manager does not support the \"all users\" member type yet.");
+      // return new Member(MemberType.ALL_USERS, null);
+    }
+
+    public static Member allAuthenticatedUsers() {
+      throw new UnsupportedOperationException("Google Cloud Resource Manager does not support the "
+          + "\"all authenticated users\" member type yet.");
+      // return new Member(MemberType.ALL_AUTHENTICATED_USERS, null);
+    }
+
+    public static Member user(String email) {
+      return new Member(MemberType.USER, email);
+    }
+
+    public static Member serviceAccount(String email) {
+      return new Member(MemberType.SERVICE_ACCOUNT, email);
+    }
+
+    public static Member group(String email) {
+      return new Member(MemberType.GROUP, email);
+    }
+
+    public static Member domain(String domain) {
+      throw new UnsupportedOperationException(
+          "Google Cloud Resource Manager does not support domain members yet.");
+      // return new Member(MemberType.DOMAIN, domain);
+    }
+
+    public MemberType type() {
+      return memberType;
+    }
+
+    public String emailOrDomain() {
+      return emailOrDomain;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(memberType, emailOrDomain);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof Member && Objects.equals(this.memberType, ((Member) obj).memberType)
+          && Objects.equals(this.emailOrDomain, ((Member) obj).emailOrDomain);
+    }
+  }
+
+  /**
+   * Represents an IAM policy binding
+   */
+  public static class Binding implements Serializable {
+
+    private static final long serialVersionUID = -8493421092718338925L;
+    private final RoleType role;
+    private final List<Member> members;
+
+    public static class Builder {
+      private RoleType role;
+      private List<Member> members;
+
+      Builder() {
+        members = new ArrayList<Member>();
+      }
+
+      public Builder role(RoleType role) {
+        this.role = role;
+        return this;
+      }
+
+      public Builder members(List<Member> members) {
+        this.members = checkNotNull(members);
```

Comment:
What about `Lists.newArrayList(checkNotNull(members))`? To avoid that `members` is changed from outside of the builder.